### DISCUSSION TITLE
Detailed Fault Definitions

### DISCUSSION THREAD

**Discussion by rahtr** [chaos-mesh/chaos-mesh#1663] [Q&A]
Hello Team,

Thanks for creating this awesome project. I have started using it and I must say that the current features and the roadmap is great.

I have a question/enhancement request. Currently we have basic definitions of the failures. Example for `pod-failure` mentioned that an error would be injected in the container but doesn't give any technical details about the fault.

![image](https://user-images.githubusercontent.com/31742059/114204637-6ad43b00-9927-11eb-8077-77aebbfd3e6a.png)

It would be great if we could compile and give a details descriptions(technical specifications) about the faults being injected or do we plan to keep it abstract?

**Comment by Colstuwjx** [comment]
IMO, as a user document, it should focus on how to get started, feature usage, and best practices. For the technical details, I think we could make a separate document that explains "how it works", e.g. ingress-nginx [how it works](https://kubernetes.github.io/ingress-nginx/how-it-works/).

**Reply by rahtr** [reply]
Sure, that's a great idea.

**Comment by AsterNighT** [comment]
Some of the chaos do have blogs/posts describing there technical details, like [iochaos](https://chaos-mesh.org/blog/how-to-simulate-io-faults-at-runtime) and [timechaos](https://chaos-mesh.org/blog/simulating-clock-skew-in-k8s-without-affecting-other-containers-on-node). But you're right, it will be great if we can present the interesting technical ideas to everyone. We are planning to make a full upgrade to our document, and help is appreciated on both user guide and technical shares.
